zenko-base: Upgrade PodDisruptionBudget to stable API

francoisferrand opened the pull request:
Migrate from `policy/v1beta1` to `policy/v1`, to support running on k8s
1.25. This changes the minimum supported k8s version to 1.21.

Issue: ZENKO-4496

francoisferrand committed:
zenko-base: Upgrade PodDisruptionBudget to stable API

Migrate from `policy/v1beta1` to `policy/v1`, to support running on k8s
1.25. This changes the minimum supported k8s version to 1.21.

Issue: ZENKO-4496

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4496 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.10`


Please check the `Fix Version/s` of ZENKO-4496, or the target
branch of this pull request.

francoisferrand committed:
zenko-base: use PodDisruptionBudget api version caps

Since there is an api version abstration already, use it to allow the
chart to target either 1.21+ or pre-1.21 k8s, depending on configuration
(or version of kubectl, by default).

Since version 1.21-1 of kubectl is used in CI, the zenko-base image will
indeed target the `policy/v1` api and require k8s 1.21+.

Issue: ZENKO-4496

KillianG commented on solution-base/mongodb/charts/mongodb/templates/poddisruptionbudget-arbiter-rs.yaml:
```diff
@@ -1,5 +1,5 @@
 {{- if and (and .Values.replicaSet.enabled .Values.replicaSet.pdb.enabled) (gt .Values.replicaSet.replicas.arbiter 0) }}
-apiVersion: policy/v1beta1
+apiVersion: policy/v1
```
Why not using an include here as well ?

francoisferrand replied:
the included-file is in the `mongodb-sharded` chart, but not in the `mongodb` chart

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4496 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.11`


Please check the `Fix Version/s` of ZENKO-4496, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4496 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.12`


Please check the `Fix Version/s` of ZENKO-4496, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

francoisferrand commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue ZENKO-4496.

Goodbye francoisferrand.
